src/Content/Version.php

```diff
@@ -186,6 +288,20 @@ public function touch(Language|string $language = 'default'): void
 	public function update(array $fields, Language|string $language = 'default'): void
 	{
 		$this->ensure($language);
-		$this->model->storage()->update($this->id, $this->language($language), $fields);
+
+		$language = Language::ensure($language);
+
+		// merge the previous state with the new state to always
+		// update to a complete version
+		$fields = [
+			...$this->read($language),
+			...$this->convertFieldNamesToLowerCase($fields)
```

Comment:
`Suggestion`: This extra lowercasing is not needed as we call `prepareFieldsBeforeWrite()` below anyway. If `array_change_key_case()` gets an array with multiple keys with the same string but different case, it automatically chooses the last one. So we can safely skip this to slightly improve performance:

```suggestion
			...$fields
```

Reply:
Oh, wow. That's super handy and good to know.